### ISSUE TITLE
Tensorboard smoothing is not smooth at all.

### ISSUE THREAD

**Issue by Ark-kun** [tensorflow/tensorflow#8829]
I have Tensorflow v1.0.1 installed on Linux.

"Smoothed" Tensorboard looks not smooth at all compared to the original graph. For some reason it manages to find some interesting loss function behavior features that simply do not exist in the original loss data.

![image](https://cloud.githubusercontent.com/assets/1829149/24485892/f269a388-14bb-11e7-8b4c-b21d54e4c4be.png)

**Comment by jubjamie** [comment]
I've found tensorboard to sometimes get a bit upset with smoothing and that I need to change the smoothing level. Can you post some screenshots of the data with 0, 0.5 and 1 smoothing?

**Comment by kbrose** [comment]
Agreed. I don't know how the smoothing is being done under the hood, but it definitely doesn't "look right".

![image](https://cloud.githubusercontent.com/assets/4053170/24511336/5cf5ef34-1531-11e7-8be4-938527377aa7.png)
(Smoothing = 1)

**Comment by jubjamie** [comment]
I believe that it uses an IIR filter as seen in the [Tensorboard Code here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tensorboard/components/vz_line_chart/vz-line-chart.ts#L455)

**Comment by kbrose** [comment]
Hmm. Does that make sense for non-periodic data? This may be too simplified but I feel like just a moving average would be better...

**Comment by aselle** [comment]
@dandelionmane, could you provide some insight, please?

**Comment by foxik** [comment]
Note that the referenced sources are in master branch, where the behaviour has much changed. In TensorFlow 1.0.1, a window average is used https://github.com/tensorflow/tensorflow/blob/r1.0/tensorflow/tensorboard/components/vz_line_chart/vz-line-chart.ts#L432 , causing the artifacts on the images.

However, an exponential moving average has been merged 4 days ago in pull request #8363, resulting in https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tensorboard/components/vz_line_chart/vz-line-chart.ts#L455.

**Comment by greaber** [comment]
It would also be nice if TensorBoard could somehow ignore outliers not just in the chart scaling but also in the smoothing.  And it would be nice if it could nonetheless let you see what those outliers are.  Since there seems to be no way to zoom in sufficiently to go step by step in TensorBoard, there is no way to find out exactly when outliers occurred and what their values are.

However, I know the long term plan for TensorBoard involves making it more pluggable and user-extensible, so it might be that the core team should just be spending its effort on making this happen so that users can supply whatever smoothing function they like or even replacements for the whole scalar display widget.

**Comment by dandelionmane** [comment]
Hi all,

I posit that no smoothing is perfect, and there are always going to be some edge cases where it doesn't behave quite like you imagine it should. 

I'm closing this issue since we're migrating to a new repo, and because it looks like the new exponential smoothing solves a lot of complaints here. If you still want smoothing changes, please open a new issue at https://github.com/tensorflow/tensorboard, with the following two things:

1. a new algorithm that you are proposing
2. examples of how the new algorithm behaves on some data sets, compared to how the current algorithm behaves
